### PR TITLE
Feature/netskope enhancement

### PR DESCRIPTION
**Pull request by TalGumi**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by tomer-pan** [comment]
Hi @TalGumi Thanks for your contribution!

A few questions:
- Why do we need a new "Netskope DLP Incident" incident type? There's already "Netskope DLP" incident type with associated fields, layout etc.
- Why do we need both "Netskope Original Severity" and "Netskope Severity" incident fields?
- Why can't we use the existing common field "External Severity" for holding the Netskope severity? We prefer using existing fields instead creating new ones if possible.
- Same question for "Netskop Original Status" -> "External Status"
- Seems like these fields are not showing up in the layout, so I'm not sure what's the purpose of mapping them:
  - Netskop Original Status
  - Netskope DLP Policy (**This one is not mapped at all in the mapper**)
  - Netskope Original Severity

File: Packs/Netskope/Integrations/NetskopeAPIv2/NetskopeAPIv2.yml [hunk 1]
```diff
@@ -3,1014 +3,1262 @@ commonfields:
...
+          description: Netskope event user key.
+          type: String
+    - arguments:
+        - description: The URL list ID to update (use netskope-url-list-list command to get URL list ID).
+          name: url_list_id
+          required: true
+        - description: The updated URL list name.
+          name: name
+          required: true
+        - description: The updated URL list items (For Exact - Enter URLs like  *.example.com, or IP addresses, separated by new line. For Regex - Enter URLs like ^client[0-9]\\.google\\.com , ^app\\.slack\\.com/.*/netskope, or ^google.com, separated by new line).
```
Comment (ShirleyDenkberg) [inline Packs/Netskope/Integrations/NetskopeAPIv2/NetskopeAPIv2.yml:979]:
```suggestion
        - description: The updated URL list items (For Exact - Enter URLs like  *.example.com, or IP addresses, separated by a new line. For Regex - Enter URLs like ^client[0-9]\\.google\\.com , ^app\\.slack\\.com/.*/netskope, or ^google.com, separated by a new line).
```

File: Packs/Netskope/Integrations/NetskopeAPIv2/NetskopeAPIv2.yml [hunk 2]
```diff
@@ -3,1014 +3,1262 @@ commonfields:
...
+          description: Netskope URL list modify type.
+          type: String
+        - contextPath: Netskope.URLList.pending
+          description: Netskope URL list pending.
+          type: Number
+    - arguments:
+        - description: The unique name for the URL list.
+          name: name
+          required: true
+        - description: The URL list items (For Exact - Enter URLs like  *.example.com, or IP addresses, separated by new line. For Regex - Enter URLs like ^client[0-9]\\.google\\.com , ^app\\.slack\\.com/.*/netskope, or ^google.com, separated by new line).
```
Comment (ShirleyDenkberg) [inline Packs/Netskope/Integrations/NetskopeAPIv2/NetskopeAPIv2.yml:1038]:
```suggestion
        - description: The URL list items (For Exact - Enter URLs like  *.example.com, or IP addresses, separated by a new line. For Regex - Enter URLs like ^client[0-9]\\.google\\.com , ^app\\.slack\\.com/.*/netskope, or ^google.com, separated by a new line).
```

File: Packs/Netskope/Integrations/NetskopeAPIv2/NetskopeAPIv2.yml [hunk 3]
```diff
@@ -3,1014 +3,1262 @@ commonfields:
...
+          description: Netskope client activate.
+          type: Boolean
+        - contextPath: Netskope.Client.emails
+          description: Netskope client emails.
+          type: String
+    - arguments:
+        - description: The URL list ID to update (use netskope-url-list-list command to get URL list ID).
+          name: url_list_id
+          required: true
+        - description: The updated URL list items (For Exact - Enter URLs like  *.example.com, or IP addresses, separated by new line. For Regex - Enter URLs like ^client[0-9]\\.google\\.com , ^app\\.slack\\.com/.*/netskope, or ^google.com, separated by new line).
```
Comment (ShirleyDenkberg) [inline Packs/Netskope/Integrations/NetskopeAPIv2/NetskopeAPIv2.yml:1185]:
```suggestion
        - description: The updated URL list items (For Exact - Enter URLs like  *.example.com, or IP addresses, separated by a new line. For Regex - Enter URLs like ^client[0-9]\\.google\\.com , ^app\\.slack\\.com/.*/netskope, or ^google.com, separated by a new line).
```

File: Packs/Netskope/ReleaseNotes/4_0_0.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
+#### Classifiers
+
+##### Netskope Classifier
+
+-   Updated the integration clarifier to support mirroring.
```
Comment (ShirleyDenkberg) [inline Packs/Netskope/ReleaseNotes/4_0_0.md:5]:
```suggestion
Updated the integration classifier to support mirroring.
```

File: Packs/Netskope/ReleaseNotes/4_0_0.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+#### Incident Types
+
+##### New: Netskope DLP Incident
+
+#### Integrations
+
+##### Netskope (API v2)
+
+-   Added the following command: **netskope-incident-list**.
```
Comment (ShirleyDenkberg) [inline Packs/Netskope/ReleaseNotes/4_0_0.md:45]:
```suggestion
-   Added the following command: ***netskope-incident-list***.
```

File: Packs/Netskope/ReleaseNotes/4_0_0.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+#### Mappers
+
+##### New: Netskope Outgoing Mapper
+
+-   Added an outgoing mapper (Available from Cortex XSOAR 6.10).
+
+##### Netskope Mapper
+
+-   Updated the Netskope mapper for DLP incident type.
```
Comment (ShirleyDenkberg) [inline Packs/Netskope/ReleaseNotes/4_0_0.md:62]:
```suggestion
New layout for Netskope DLP incidents (Available from Cortex XSOAR 6.10).

#### Mappers

##### New: Netskope Outgoing Mapper

Added an outgoing mapper (Available from Cortex XSOAR 6.10).

##### Netskope Mapper

-Updated the Netskope mapper for the DLP incident type.
```

File: Packs/Netskope/Integrations/NetskopeAPIv2/README.md [hunk 7]
```diff
@@ -43,135 +43,138 @@ Retrieve alerts generated by Netskope. Select desired alerts using the alert_typ
...
+| Netskope.Alert.src_epoch_now             | Number   | A numeric value representing the source epoch.                                                                                                                                                                                                                                                                                                                                |
+| Netskope.Alert.access_method             | String   | Cloud app traffic can be steered to the Netskope cloud using different deployment methods such as Client (Netskope Client), Secure Forwarder etc. Administrators can also upload firewall and/or proxy logs for log analytics. This field shows the actual access method that triggered the event. For log uploads this shows the actual log type such as PAN, Websense, etc. |
```
Comment (ShirleyDenkberg) [inline Packs/Netskope/Integrations/NetskopeAPIv2/README.md:81]:
```suggestion
| Netskope.Alert.access_method             | String   | Cloud app traffic can be steered to the Netskope cloud using different deployment methods such as Client (Netskope Client), Secure Forwarder etc. Administrators can also upload firewall and/or proxy logs for log analytics. This field shows the actual access method that triggered the event. For log uploads, this shows the actual log type such as PAN, Websense, etc. |
```

File: Packs/Netskope/Integrations/NetskopeAPIv2/README.md [hunk 8]
```diff
@@ -43,135 +43,138 @@ Retrieve alerts generated by Netskope. Select desired alerts using the alert_typ
...
+| Netskope.Alert.dst_timezone              | String   | Destination timezone.                                                                                                                                                                                                                                                                                                                                                         |
+| Netskope.Alert.dst_zipcode               | String   | Application’s zip code as determined by Maxmind or IP2Location Geodatabase.                                                                                                                                                                                                                                                                                                   |
```
Comment (ShirleyDenkberg) [inline Packs/Netskope/Integrations/NetskopeAPIv2/README.md:108]:
```suggestion
| Netskope.Alert.dst_country               | String   | Application’s two-letter country code as determined by MaxMind or IP2Location Geolocation database.                                                                                                                                                                                                                                                                                    |
| Netskope.Alert.dst_latitude              | Number   | Latitude of the application as determined by MaxMind or IP2Location Geolocation database.                                                                                                                                                                                                                                                                                              |
| Netskope.Alert.dst_location              | String   | Application’s city as determined by MaxMind or IP2Location Geolocation database.                                                                                                                                                                                                                                                                                                       |
| Netskope.Alert.dst_longitude             | Number   | Longitude of the application as determined by MaxMind or IP2Location Geolocation database.                                                                                                                                                                                                                                                                                             |
| Netskope.Alert.dst_region                | String   | Application’s state or region as determined by MaxMind or IP2Location Geolocation database.                                                                                                                                                                                                                                                                                            |
| Netskope.Alert.dst_timezone              | String   | Destination timezone.                                                                                                                                                                                                                                                                                                                                                         |
| Netskope.Alert.dst_zipcode               | String   | Application’s zip code as determined by MaxMind or IP2Location Geolocation database.                                                                                                                                                                                                                                                                                                   |
```

File: Packs/Netskope/Integrations/NetskopeAPIv2/README.md [hunk 9]
```diff
@@ -43,135 +43,138 @@ Retrieve alerts generated by Netskope. Select desired alerts using the alert_typ
...
+| Netskope.Alert.src_timezone              | String   | Source timezone. Shows the long-format timezone designation.                                                                                                                                                                                                                                                                                                                  |
+| Netskope.Alert.src_zipcode               | String   | Source zip code as determined by Maxmind or IP2Location Geodatabase.                                                                                                                                                                                                                                                                                                          |
```
Comment (ShirleyDenkberg) [inline Packs/Netskope/Integrations/NetskopeAPIv2/README.md:139]:
```suggestion
| Netskope.Alert.src_country               | String   | User’s country’s two-letter country code as determined by MaxMind or IP2Location Geolocation database.                                                                                                                                                                                                                                                                                 |
| Netskope.Alert.src_latitude              | Number   | Latitude of the user as determined by MMaxMind or IP2Location Geolocation database.                                                                                                                                                                                                                                                                                                     |
| Netskope.Alert.src_location              | String   | User’s city as determined by MaxMind or IP2Location Geolocation database.                                                                                                                                                                                                                                                                                                              |
| Netskope.Alert.src_longitude             | Number   | Longitude of the user as determined by MaxMind or IP2Location Geolocation database.                                                                                                                                                                                                                                                                                                    |
| Netskope.Alert.src_region                | String   | Source state or region as determined by MaxMind or IP2Location Geolocation database.                                                                                                                                                                                                                                                                                                   |
| Netskope.Alert.src_time                  | Date     | A timestamp associated with the source or event.                                                                                                                                                                                                                                                                                                                              |
| Netskope.Alert.src_timezone              | String   | Source timezone. Shows the long-format timezone designation.                                                                                                                                                                                                                                                                                                                  |
| Netskope.Alert.src_zipcode               | String   | Source zip code as determined by MaxMind or IP2Location Geolocation database.                                                                                                                                                                                                                                                                                                          |
```

File: Packs/Netskope/Integrations/NetskopeAPIv2/README.md [hunk 10]
```diff
@@ -43,135 +43,138 @@ Retrieve alerts generated by Netskope. Select desired alerts using the alert_typ
...
+| Netskope.Alert.nsp_rtt_back              | Number   | Round-trip time information for NSP on the back end.                                                                                                                                                                                                                                                                                                                          |
+| Netskope.Alert.nsp_rtt_front             | Number   | Round-trip time information for NSP on the front end.                                                                                                                                                                                                                                                                                                                         |
```
Comment (ShirleyDenkberg) [inline Packs/Netskope/Integrations/NetskopeAPIv2/README.md:159]:
```suggestion
| Netskope.Alert.nsp_rtt_back              | Number   | Roundtrip time information for NSP on the back end.                                                                                                                                                                                                                                                                                                                          |
| Netskope.Alert.nsp_rtt_front             | Number   | Roundtrip time information for NSP on the front end.                                                                                                                                                                                                                                                                                                                         |
```

File: Packs/Netskope/Integrations/NetskopeAPIv2/README.md [hunk 11]
```diff
@@ -529,31 +520,34 @@ Update the URL List with the values provided. please note that this command over
...
+| ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------------ |
+| url_list_id       | The URL list ID to update (use netskope-url-list-list command to get URL list ID).                                                                                                                                                               | Required     |
+| name              | The updated URL list name.                                                                                                                                                                                                                       | Required     |
+| urls              | The updated URL list items (For Exact - Enter URLs like _.example.com, or IP addresses, separated by new line. For Regex - Enter URLs like ^client[0-9]\\.google\\.com , ^app\\.slack\\.com/._/netskope, or ^google.com, separated by new line). | Required     |
```
Comment (ShirleyDenkberg) [inline Packs/Netskope/Integrations/NetskopeAPIv2/README.md:527]:
```suggestion
| urls              | The updated URL list items (For Exact - Enter URLs like _.example.com, or IP addresses, separated by a new line. For Regex - Enter URLs like ^client[0-9]\\.google\\.com , ^app\\.slack\\.com/._/netskope, or ^google.com, separated by a new line). | Required     |
```

File: Packs/Netskope/Integrations/NetskopeAPIv2/README.md [hunk 12]
```diff
@@ -593,30 +586,33 @@ Create a new URL list.
...
+| **Argument Name** | **Description**                                                                                                                                                                                                                          | **Required** |
+| ----------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------ |
+| name              | The unique name for the URL list.                                                                                                                                                                                                        | Required     |
+| urls              | The URL list items (For Exact - Enter URLs like _.example.com, or IP addresses, separated by new line. For Regex - Enter URLs like ^client[0-9]\\.google\\.com , ^app\\.slack\\.com/._/netskope, or ^google.com, separated by new line). | Required     |
```
Comment (ShirleyDenkberg) [inline Packs/Netskope/Integrations/NetskopeAPIv2/README.md:592]:
```suggestion
| urls              | The URL list items (For Exact - Enter URLs like _.example.com, or IP addresses, separated by a new line. For Regex - Enter URLs like ^client[0-9]\\.google\\.com , ^app\\.slack\\.com/._/netskope, or ^google.com, separated by a new line). | Required     |
```

File: Packs/Netskope/Integrations/NetskopeAPIv2/README.md [hunk 13]
```diff
@@ -951,24 +928,297 @@ Update the URL list with the values provided. Note that this command appends the
 
 #### Input
 
+| **Argument Name** | **Description**                                                                                                                                                                                                                                  | **Required** |
+| ----------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------------ |
+| url_list_id       | The URL list ID to update (use netskope-url-list-list command to get URL list ID).                                                                                                                                                               | Required     |
+| urls              | The updated URL list items (For Exact - Enter URLs like _.example.com, or IP addresses, separated by new line. For Regex - Enter URLs like ^client[0-9]\\.google\\.com , ^app\\.slack\\.com/._/netskope, or ^google.com, separated by new line). | Required     |
```
Comment (ShirleyDenkberg) [inline Packs/Netskope/Integrations/NetskopeAPIv2/README.md:934]:
```suggestion
| urls              | The updated URL list items (For Exact - Enter URLs like _.example.com, or IP addresses, separated by a new line. For Regex - Enter URLs like ^client[0-9]\\.google\\.com , ^app\\.slack\\.com/._/netskope, or ^google.com, separated by a new line). | Required     |
```

**Comment by ShirleyDenkberg** [comment]
@thefrieddan1 @tomer-pan Doc review completed.

**Comment by TalGumi** [comment]
Hi @tomer-pan ,
Thank you for your review.

About your questions:
- In fetch-incidents command, we fetch alerts, events, and incidents. There are DLP alerts (Netskope DLP) and DLP incidents (Netskope DLP Incident).
- For mirroring, we need both "Netskope Original Severity" and "Netskope Severity" incident fields. The Netskope API requires old values and new values to update status and severity fields. "Netskope Original Severity" holds the old severity value. and "Netskope Severity" holds the new severity value.
- Netskope severity has different predefined values from the existing common field "External Severity" predefined values. That's why we don't use the existing common field "External Severity" for holding the Netskope severity.
- We defined Netskope status as a free text field and the existing common field "External Status" is a drop-down field with predefined values.
- The fields 'Netskop Original Status' and 'Netskop Original Severity' are not shown in the layout cause they are used for mirroring (updating field value according to old value). About 'Netskope DLP Policy' - I will check with our security engineer 
the need for this field and will update you soon.

**Review by thefrieddan1 — APPROVED** [review]

**Pull request merged**